src/thorin/plug/direct/pass/cps2ds.cpp

```diff
@@ -129,12 +129,25 @@ const Def* CPS2DS::rewrite_body_(const Def* def) {
     }
 
     if (def->isa<Var>()) return def;
+    if (def->isa<Global>()) return def;
 
     if (auto tuple = def->isa<Tuple>()) {
         auto elements = DefVec(tuple->ops(), [&](const Def* op) { return rewrite_body(op); });
         return world().tuple(def->type(), elements)->set(tuple->dbg());
     }
 
+    // TODO there are more probls like this
+    if (auto old_mut = def->isa_mut()) {
```

Comment:
Is there a high-level view of what is going on here/what the problem is
for the next one reading this code?